### PR TITLE
fix #1719

### LINKED ISSUE #1719: Several possible memory leak defects.
Hello! Today, I rechecked nanomq using the memory leak detection tool I developed and found several potential memory leak defects (there may be false positives). I conducted preliminary filtering of the error reports and uploaded the filtered reports of potentially genuine memory leak defects to the following link: https://gitee.com/yin-luming/nanomq_full_report.git. Please note that there are multiple HTML files in the repository, and one HTML file may contain reports for multiple defects (click on the Reports tab on the left side of the HTML page to switch views). When confirming, simply focus on defects labeled as memory leaks; almost all other types of defects are likely false positives. However, some memory leak types of defects may also be false positives. You only need to pull the repository to a Windows operating system and use a browser to view the reports.

### PR DESCRIPTION
**Pull request by OdyWayne**

### CODE REVIEW

File: nanomq/webhook_post.c
Comment (OdyWayne) [inline nanomq/webhook_post.c:1]: for line 546 and line 554, if `parquet` and `blf` are both enabled then `flush_smsg_to_disk` will be called twice, but the `msgs_del` will be freed in the first called, then there will be a u-a-f. What is the business logic here，should we fix it? @xinyi-xs

File: nanomq/webhook_inproc.c [hunk 1]
```diff
@@ -744,6 +744,7 @@ hook_cb(void *arg)
 		nng_msleep(3600000); // neither pause() nor sleep() portable
 	}
 
+out:
```
[the remark this answers is not in the document]
Comment (OdyWayne) [reply nanomq/webhook_inproc.c:747]: `hook_search_reset_aio` is a global variable and is already decleared. `works` **is** alloced before line 714 in line 707.

File: nanomq/cmd_proc.c [hunk 2]
```diff
@@ -92,6 +92,7 @@ handle_recv(const char *msg, size_t msg_len, conf *config, char **err_msg)
 	return 0;
 
 err:
+	conf_fini(new_conf);
```
[the remark this answers is not in the document]
Comment (OdyWayne) [reply nanomq/cmd_proc.c:95]: Yes ur right, it's a false positive.

File: nanomq/webhook_inproc.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (wanghaEMQ) [reply nanomq/webhook_inproc.c:747]: Are u kidding me? Where will the works[0] be allocated?

**Review by wanghaEMQ — CHANGES_REQUESTED** [review]

File: nanomq/cmd_proc.c (hunk 2 above)
[the remark this answers is not in the document]
Comment (wanghaEMQ) [reply nanomq/cmd_proc.c]: So... Fix it in another way?

File: nanomq/webhook_inproc.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (OdyWayne) [reply nanomq/webhook_inproc.c:747]: compiler can deal with this as long as it is declared before.

File: nanomq/cmd_proc.c (hunk 2 above)
[the remark this answers is not in the document]
Comment (OdyWayne) [reply nanomq/cmd_proc.c]: > So... Fix it in another way?

just ignore it, it's a **false positive**.

File: nanomq/webhook_inproc.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (OdyWayne) [reply nanomq/webhook_inproc.c:748]: as you wish

**Review by JaylinYu — APPROVED** [review]

File: nanomq/webhook_post.c
Comment (JaylinYu) [reply nanomq/webhook_post.c:1]: for now, yes.
But might be 2 MQTT Stream object exists together in the future.

**Comment by JaylinYu** [comment]
BLF & Parquet shall not be enabled together for now.

**Review by wanghaEMQ — APPROVED** [review]

**Pull request merged**